.github/workflows/build_main.yml

```diff
@@ -1,168 +1,200 @@
+
+    ###############################
+    # SETUP ENVIRONMENT IF NEEDED #
+    ###############################
+    - name: Set env var 1
+      if: ${{ inputs.env_var_name_1 != '' }}
+      run: |
+        echo "::add-mask::${{ secrets.env_var_value_1 }}"
+        echo "${{ inputs.env_var_name_1 }}=${{ secrets.env_var_value_1 }}" >> $env:GITHUB_ENV
+        
+    - name: Set env var 2
+      if: ${{ inputs.env_var_name_2 != '' }}
+      run: |
+        echo "::add-mask::${{ secrets.env_var_value_2 }}"
+        echo "${{ inputs.env_var_name_2 }}=${{ secrets.env_var_value_2 }}" >> $env:GITHUB_ENV
+        
+    - name: Set env var 3
+      if: ${{ inputs.env_var_name_3 != '' }}
+      run: |
+        echo "::add-mask::${{ secrets.env_var_value_3 }}"
+        echo "${{ inputs.env_var_name_3 }}=${{ secrets.env_var_value_3 }}" >> $env:GITHUB_ENV
+        
+    - name: Set env var 4
+      if: ${{ inputs.env_var_name_4 != '' }}
+      run: |
+        echo "::add-mask::${{ secrets.env_var_value_4 }}"
+        echo "${{ inputs.env_var_name_4 }}=${{ secrets.env_var_value_4 }}" >> $env:GITHUB_ENV
+
+    - name: Set env var 5
+      if: ${{ inputs.env_var_name_5 != '' }}
+      run: |
+        echo "::add-mask::${{ secrets.env_var_value_5 }}"
+        echo "${{ inputs.env_var_name_5 }}=${{ secrets.env_var_value_5 }}" >> $env:GITHUB_ENV
+
+    - name: Set env var 6
+      if: ${{ inputs.env_var_name_6 != '' }}
+      run: |
+        echo "::add-mask::${{ secrets.env_var_value_6 }}"
+        echo "${{ inputs.env_var_name_6 }}=${{ secrets.env_var_value_6 }}" >> $env:GITHUB_ENV
+
+    - name: Set working directory if provided
+      if: ${{ inputs.workdir != '' }}
+      run: |
+        echo "workdir=${{ inputs.workdir }}" >> $env:GITHUB_ENV
+
+    - name: Set working directory to default if not provided
+      if: ${{ inputs.workdir == '' }}
+      run: |
+        echo "workdir=." >> $env:GITHUB_ENV
+        
+    ###############################
+    # Run prebuild command        #
+    ###############################
+    - name: Run prebuild command
+      if: ${{ inputs.prebuild_command != '' }}
+      run: ${{ inputs.prebuild_command }}
+      working-directory: "${{ env.workdir }}"
+      
+    ###############################
+    # Build, test, report results #
+    ###############################
+    - name: Build
+      run: dotnet build
+      working-directory: "${{ env.workdir }}"
+    
+    - name: Test
+      run: dotnet test --collect:"XPlat Code Coverage"
+      working-directory: "${{ env.workdir }}"
+
+    - name: Check Code Coverage
+      id: check_code_coverage
+      shell: pwsh
+      run: |
+        Get-ChildItem -Path "." -Include "coverage.cobertura.xml" -Recurse | Copy-Item -Destination .
+        [double] $coverage = ([xml]($tmp = Get-Content -Path 'coverage.cobertura.xml')).SelectSingleNode("/coverage/packages/package[@name='${{ env.workdir }}']").Attributes['line-rate'].Value
```

Comment:
Does this disable the use of forward slash in workdir?

Reply:
I think that according to all our guidelines we should not have any slashes (ie sobfolders) in working dir though?

Reply:
The reason for the need for forward slashes is referring to the issue with Code Quality Check using the workdir to determine what to check inside the repository. When we have multiple similarly starting Tasks the Code Quality Check tries to check every folder with that filter. For example Frends.HTTP.Request and Frends.HTTP.RequestBytes. Of course I can try to fix this in the workflow itself.